[E2E/Acceptance CI Failure]: Acceptance (interested-partner-organization/view-the-blog)

### I have read the procedure

- [ ] I reproduced the flake on the develop branch.
- [ ] I observed this flake directly on the develop branch.
- [x] The changes in my PR are unrelated to the failure.

### CI Test Type

Acceptance

### Which CI step failed?

Run Mobile Acceptance Test interested-partner-organization/view-the-blog

### Did the test fail after an automatic rerun?

Test failed once and then passed after an automatic rerun

### Did the test fail when running on desktop, mobile, or both?

Mobile

### Stacktrace

```shell
FAIL core/tests/puppeteer-acceptance-tests/specs/interested-partner-organization/view-the-blog.spec.ts (422.864 s)
  Logged-out User
    ✓ should observe the welcome message and empty-state notice when no blog posts exist (3446 ms)
    ✓ should observe blogs with navigation (36933 ms)
    ✓ should paginate blog posts after the first 10 items (205334 ms)
    ✓ should display blog page with correct layout elements (4185 ms)
    ✓ should display blog post page with all required elements (3725 ms)
    ✕ should be able to search and filter blog posts (11615 ms)

  ● Logged-out User › should be able to search and filter blog posts

    No results found containing "International"

      1058 |
      1059 |     if (!contentFound) {
    > 1060 |       throw new Error(`No results found containing "${text}"`);
           |             ^
      1061 |     }
      1062 |   }
      1063 |

      at LoggedOutUser.expectBlogSearchResultsToContain (utilities/user/logged-out-user.ts:1060:13)
          at runMicrotasks (<anonymous>)
      at Object.<anonymous> (specs/interested-partner-organization/view-the-blog.spec.ts:154:7)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 5 passed, 6 total
Snapshots:   0 total
Time:        424.426 s, estimated 456 s
Ran all test suites matching /core\/tests\/puppeteer-acceptance-tests\/specs\/interested-partner-organization\/view-the-blog.spec.ts/i.
INFO     2026-04-06 10:29:50,416 shutdown.py:48] Shutting down.
Error: 4-06 10:29:50 +0000] [5872] [ERROR] Worker (pid:5898) was sent SIGTERM!
Error: 4-06 10:29:50 +0000] [6087] [ERROR] Worker (pid:6090) was sent SIGTERM!
Error: 4-06 10:29:50 +0000] [6057] [ERROR] Worker (pid:6061) was sent SIGTERM!
Error: 4-06 10:29:50 +0000] [6491] [ERROR] Worker (pid:6537) was sent SIGTERM!
INFO     2026-04-06 10:29:51,265 stub_util.py:321] Applying all pending transactions and saving the datastore
INFO     2026-04-06 10:29:51,265 stub_util.py:324] Saving search indexes
                                                              
 
i  emulators: Received SIGTERM for the first time. Starting a clean shutdown.
i  emulators: Please wait for a clean shutdown or send the SIGTERM signal again to stop right now.
i  Automatically exporting data using --export-on-exit "/home/runner/work/oppia/oppia/../firebase_emulator_cache" please wait for the export to finish...
⚠  Emulator UI has exited upon receiving signal: SIGTERM
i  Found running emulator hub for project dev-project-id at http://localhost:4400
i  Exporting data to: /home/runner/work/oppia/firebase_emulator_cache
i  emulators: Received export request. Exporting data to /home/runner/work/oppia/firebase_emulator_cache.
✔  emulators: Export complete.
✔  Export complete
i  emulators: Shutting down emulators.
i  ui: Stopping Emulator UI
⚠  ui: Error stopping Emulator UI
i  auth: Stopping Authentication Emulator
i  hub: Stopping emulator hub
i  logging: Stopping Logging Emulator

ERROR:root:Portserver failed to shut down after 10 seconds.
```

### Screenshots and/or Screen Recordings

Failure point snapshots
<img width="750" height="1334" alt="Image" src="https://github.com/user-attachments/assets/f0db0188-41c4-4752-9492-e24506ef977c" />
<img width="750" height="1334" alt="Image" src="https://github.com/user-attachments/assets/54989e7e-faba-4302-aecc-ffc41fbf88dd" />
<img width="750" height="1334" alt="Image" src="https://github.com/user-attachments/assets/0593e9fb-21f1-4931-9c08-9de527e2f50b" />

Video Recordings

https://github.com/user-attachments/assets/598cfcd9-9ebd-41c8-ab3b-181c27fd4049
https://github.com/user-attachments/assets/bfb19e6b-0430-487f-8ae4-d176e9c65eb1
https://github.com/user-attachments/assets/5bb6f5b3-8c6e-4032-ab53-52b7e75e74b6

### Occurrences

https://github.com/oppia/oppia/actions/runs/24027068981/job/70069654706?pr=25419

### Additional Information

_No response_

### Debugging document link

_No response_

--- a/core/tests/puppeteer-acceptance-tests/utilities/user/logged-out-user.ts
+++ b/core/tests/puppeteer-acceptance-tests/utilities/user/logged-out-user.ts
@@ -1045,6 +1045,9 @@ export class LoggedOutUser extends BaseUser {
     if (!blogPostsFound) {
       return;
     }
+    await this.page.waitForSelector(blogPostContentSelector, {
+      visible: true,
+    });
     const contentFound = await this.page.$$eval(
       `${blogPostTitleContainerSelector}, ${blogPostContentSelector}`,
       (elements, searchText) =>
@@ -1093,6 +1096,9 @@ export class LoggedOutUser extends BaseUser {
     if (!blogPostsFound) {
       return;
     }
+    await this.page.waitForSelector(blogPostTagSelector, {
+      visible: true,
+    });
     const tagFound = await this.page.$$eval(
       blogPostTagSelector,
       (elements, expectedTag) =>

--- a/core/tests/puppeteer-acceptance-tests/utilities/user/logged-out-user.ts
+++ b/core/tests/puppeteer-acceptance-tests/utilities/user/logged-out-user.ts
@@ -1045,9 +1045,7 @@ export class LoggedOutUser extends BaseUser {
     if (!blogPostsFound) {
       return;
     }
-    await this.page.waitForSelector(blogPostContentSelector, {
-      visible: true,
-    });
+    await this.expectElementToBeVisible(blogPostContentSelector);
     const contentFound = await this.page.$$eval(
       `${blogPostTitleContainerSelector}, ${blogPostContentSelector}`,
       (elements, searchText) =>
@@ -1096,9 +1094,7 @@ export class LoggedOutUser extends BaseUser {
     if (!blogPostsFound) {
       return;
     }
-    await this.page.waitForSelector(blogPostTagSelector, {
-      visible: true,
-    });
+    await this.expectElementToBeVisible(blogPostTagSelector);
     const tagFound = await this.page.$$eval(
       blogPostTagSelector,
       (elements, expectedTag) =>

